### ISSUE TITLE
[BUG] Delete Topics doesn't throw PulsarAdminException or ERROR

### ISSUE THREAD

**Issue by zhouxy0809** [streamnative/kop#1328]
**Describe the bug**
Kop just throw ERROR.UNKNOWN_TOPIC_OR_PARTITION when deleting topics，not throw PulsarAdminException or Error，is not friendly to upper client.

**To Reproduce**
1.  Frequently thow  ERROR “This server does not host this topic-partition.”  when executing openmessaing benchmark kafka drivers；
2.  Check kop log, and find  perform http request of PuslarClientAdmin deleteTopics to response HTTP 500 Internal Server Error
3.  Check pulsar log, and find Pulsar delete topic with different type exception, such as "Topic has active producers/subscriptions" or "Failed to delete authentication policies for partitioned topic" and so on.

**Expected behavior**
KOP could be throw pulsar exception or error to client.

**Screenshots**
![image](https://user-images.githubusercontent.com/19686515/172984601-281a66f7-2050-4ddb-b2c0-16c90a40eb67.png)
![image](https://user-images.githubusercontent.com/19686515/172984892-e178d4ea-1e1e-40d1-921c-b7e1aa836769.png)
![image](https://user-images.githubusercontent.com/19686515/172985850-610f2cae-cfef-4183-9b05-cb7371769ef6.png)
![image](https://user-images.githubusercontent.com/19686515/172985914-066ab17e-b0b7-4bb0-897b-0c77cada347d.png)

**Additional context**
KOP VERSION： 2.9.2.8

**Comment by BewareMyPower** [comment]
It can be reproduced by running a producer at first (don't exit):

```bash
$ ./bin/kafka-console-producer.sh --bootstrap-server localhost:9092 --topic my-topic
>hello
>

```

Then try to delete the topic:

```bash
$ ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --delete --topic my-topic
Error while executing topic command : This server does not host this topic-partition.
[2022-06-10 22:53:31,139] ERROR org.apache.kafka.common.errors.UnknownTopicOrPartitionException: This server does not host this topic-partition.
```

**Comment by BewareMyPower** [comment]
> KOP could be throw pulsar exception or error to client.

No. The error is printed by Kafka's admin client because it received the `UNKNOWN_TOPIC_OR_PARTITION` error code from KoP when the topic failed to be deleted. Kafka's admin client can only receive the [error code](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/protocol/Errors.java), so throwing a `PulsarAdminException` at KoP side is meaningless.

**Comment by zhouxy0809** [comment]
> > KOP could be throw pulsar exception or error to client.
> 
> No. The error is printed by Kafka's admin client because it received the `UNKNOWN_TOPIC_OR_PARTITION` error code from KoP when the topic failed to be deleted. Kafka's admin client can only receive the [error code](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/protocol/Errors.java), so throwing a `PulsarAdminException` at KoP side is meaningless.

In my opinion，Kop just protocol handler between Kafka application and pulsar cluster，it's essentially to server to pulsar, and user must be skilled in pulsar operating。It's more user friendly if  Kop could be provide the extension of pulsar exception, personal opinion just for your reference。

**Comment by eolivelli** [comment]
In our fork of OpenMessaging Benchmarks (datastax/openmessaging-benchmarks) we added some code to ignore the error.
It happens quite often when you have multiple worker, because each worker runs the same code

**Comment by BewareMyPower** [comment]
In short, there is no way for Kafka's `AdminClient` to print any error message. It's a limitation of the **Kafka client.**

How to make it more friendly? Do you mean to make Kafka's client print some messages like "Topic has active producers/subscriptions"?

The only way is to change Kafka's protocol and modify Kafka's source code. KoP acts as a handler to implement Kafka's protocols. Unfortunately, Kafka's protocols don't define a way to accept any string when the delete operation failed.
